openpype/hosts/maya/api/plugin.py

```diff
@@ -405,14 +405,21 @@ def collect_instances(self):
                 # No existing scene instance node for this layer. Note that
                 # this instance will not have the `instance_node` data yet
                 # until it's been saved/persisted at least once.
-                # TODO: Correctly define the subset name using templates
-                prefix = self.layer_instance_prefix or self.family
-                subset_name = "{}{}".format(prefix, layer.name())
+                project_name = self.create_context.get_current_project_name()
                 instance_data = {
-                    "asset": legacy_io.Session["AVALON_ASSET"],
-                    "task": legacy_io.Session["AVALON_TASK"],
+                    "asset": self.create_context.get_current_asset_name,
+                    "task": self.create_context.get_current_task_name,
                     "variant": layer.name(),
                 }
+                asset_doc = get_asset_by_name(project_name,
+                                              instance_data["asset"])
+                subset_name = get_subset_name(
```

Comment:
Shouldn't this be using ]`self.get_subset_name()`](https://github.com/ynput/OpenPype/blob/develop/openpype/pipeline/create/creator_plugins.py)?

Reply:
`self.get_subset_name` uses `self.family` which is `renderLayer` and not `render` as I would expect. 

Situation with families in Maya (`rendering`, `renderLayer`, `render`) is confusing to say the least.

Reply:
Shouldn't be in that case overriden the method to use right value, so if the method is called out of the inner plugin logic you'll get the same result.

Reply:
Yup, like @iLLiCiTiT mentions. If it returns the wrong value then I suspect we override it on that creator plugin so it returns the right value instead. (There might be other things calling that method on the Creator and expecting the correct subset name!)

Reply:
Ok, modified.